src/com/dmdirc/logger/ErrorManager.java

```diff
@@ -312,24 +303,9 @@ public void deleteError(final ProgramError error) {
      * @since 0.6.3m1
      */
     public void deleteAll() {
-        final Set<ProgramError> errorsCopy;
-        synchronized (errors) {
-            errorsCopy  = Sets.newHashSet(errors);
-            errors.clear();
-        }
-        errorsCopy.forEach(e -> {
-            fireErrorDeleted(e);
-            eventBus.publish(new ProgramErrorDeletedEvent(e));
-        });
-    }
-
-    /**
-     * Returns the number of errors.
-     *
-     * @return Number of ProgramErrors
-     */
-    public int getErrorCount() {
-        return errors.size();
+        final Set<ProgramError> errorsCopy = Sets.newHashSet(errors);
+        errors.clear();
+        errorsCopy.forEach(e -> eventBus.publish(new ProgramErrorDeletedEvent(e)));
```

Comment:
nit: `errorsCopy.map(ProgramErrorDeletedEvent::new).forEach(eventBus::publish)` would get rid of the lambda